src/Symfony/Component/Mailer/Transport/Dsn.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Transport;
+
+use Symfony\Component\Mailer\Exception\InvalidArgumentException;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+class Dsn
+{
+    private $scheme;
+
+    private $host;
+
+    private $port;
+
+    private $user;
+
+    private $pass;
+
+    private $options;
+
+    public function __construct(string $scheme, string $host, ?int $port, string $user, string $pass, array $options = [])
+    {
```

Comment:
Port argument optional to maybe move it to the end?

Reply:
Looks more natural like this.